dev-libs/openssl/openssl102-1.0.2m.recipe

```diff
@@ -18,36 +18,53 @@ LICENSE="OpenSSL"
 REVISION="1"
 SOURCE_URI="https://www.openssl.org/source/openssl-$portVersion.tar.gz"
 CHECKSUM_SHA256="8c6ff15ec6b319b50788f42c7abc2890c08ba5a1cdcd3810eb9092deada37b0f"
+SOURCE_DIR="openssl-$portVersion"
 PATCHES="openssl-$portVersion.patchset"
 
 ARCHITECTURES="x86_gcc2 x86 x86_64 ?arm ?ppc"
 SECONDARY_ARCHITECTURES="x86_gcc2 x86"
 
 PROVIDES="
-	openssl$secondaryArchSuffix = $portVersion
-	lib:libcrypto$secondaryArchSuffix = 1.0.0 compat >= 1.0.0
-	lib:libssl$secondaryArchSuffix = 1.0.0 compat >= 1.0.0
```

Comment:
What's the point of changing the recipe name? The lib version?

Reply:
Well, if we put 1.0.2 in openssl102 and 1.1 in openssl11 then we have a consistent naming convention and we can even add a wrapper package ("openssl") which would depend on the recommended openssl version, i.e. openssl102 for the moment, and then openssl11 once we have a working package for 1.1.
BTW, it might even be a good idea to add the symlinks (discussed above) in the openssl package.
I'm quite sure this will even simplify the transition of the packages which depend on libssl and/or libcrypto. Later, when all of these packages get bumped, I guess we could even drop the symlinks and only keep an enpty openssl package with just a dependency on our recommened/default openssl.

Reply:
@korli, you are right. There's probably no need to change the package name. We can keep openssl for 1.0.2.
We could then add 1.1.0 in openssl11, keeping 1.0.2 in openssl.
Later we could rename openssl to openssl102 and openssl11 to openssl.

Reply:
openssl11 already exists AFAICT. Whether openssl 1.1 and openssl 1.0 can coexist in the runtime is a good question. Typically what happens when two libraries in one application each use one different version of openssl?

Reply:
My mistake. I know openssl11 already exists. I should have written "enable" openssl11 instead of "add" openssl11.
BTW, I forgot to say that we would/will need to change openssl11 a little to have it provide `libssl.so.1.1.0` and `libcrypto.so.1.1.0` and also move `lib/engines/` to `lib/openssl-1.1/engines/`.

Hmm, I've just noticed that Debian uses `$libDir/openssl-1.0.2/engines/` for 1.0.2 and `$libDir/engines-1.1/` for 1.1.
So I think it would be nicer to use `$libDir/engines-1.0.2/` for 1.0.2 and `$libDir/engines-1.1/` for 1.1.

I'll do some tests and check whether or not it is possible to install openssl 1.0.2 (with the new layout and with the symlinks) without having to uninstall all the packages which depend on libssl and/or libcrypto.
I'll also need to check that building packages against 1.0.2 (with the new layout) only adds dependencies to `libssl.so.1.0.2` and `libcrypto.so.1.0.2` (and not to `libssl.so.1.0.0` and `libcrypto.so.1.0.0`, which will be symlinks).

Reply:
> Typically what happens when two libraries in one application each use one different version of openssl?

I believe this would be OK because if two (or more) libs (required by an app) depend on the libs of different versions of openssl, then this means they use these openssl libs in different contexts.
All we need to do is make sure the base packages for 1.0.2 and 1.1.0 only ship the libs in openssl (and not the `cmd:openssl` runtime), and that they ship them with distinct sonames. That's what Debian does.

For anyone else catching this thread:
openssl11 would ship `lib{ssl,crypto}.so.1.1.0` and openssl102 (or openssl if we keep the package name unchanged) would ship `lib{ssl,crypto}.so.1.0.2`.
openssl11_bin would ship `cmd:openssl = 1.1.0` and openssl102_bin (or openssl_bin) would ship `cmd:openssl = 1.0.2`. So each user would be able to install one (and only one) cmd:openssl from the corresponding _bin sub-package.